addons/mail/models/mail_thread.py

```diff
@@ -808,6 +809,109 @@ def _routing_reset_bounce(self, email_message, message_dict):
+        """
+        email_from = message_dict.get('email_from')
+        if not email_from:
+            return False
+
+        email_from_normalized = tools.email_normalize(email_from)
+
+        # Detect the email address sent to many emails
+        get_param = self.env['ir.config_parameter'].sudo().get_param
+        INCOMING_LIMIT_PERIOD = int(get_param('mail.incoming.limit.period', 120))
+        INCOMING_LIMIT_ALIAS = int(get_param('mail.incoming.limit.alias', 20))
+
+        create_date_limit = datetime.datetime.now() - datetime.timedelta(minutes=INCOMING_LIMIT_PERIOD)
+
+        # Search only once per model
+        models = {
+            self.env[model]
+            for model, thread_id, *__ in routes or []
+            if not thread_id  # Reply to an existing thread
+        }
+
+        for model in models:
+            if not hasattr(model, '_routing_detect_loop_from_records_domain'):
+                continue
+
+            domain = model._routing_detect_loop_from_records_domain(email_from_normalized)
+
+            if not domain:
+                _logger.info('Domain missing for %s, could not find duplicated.', self._name)
+                continue
+
+            mail_incoming_messages_count = model.sudo().search_count(
+                expression.AND([
+                    [('create_date', '>', create_date_limit)],
+                    domain,
+                ]),
+            )
+
+            if mail_incoming_messages_count >= INCOMING_LIMIT_ALIAS:
+                _logger.info('This email address created too many <%s>.', model)
+
+                body = self.env['ir.qweb']._render(
+                    'mail.message_notification_limit_email',
+                    {'email': message_dict.get('to')},
+                    minimal_qcontext=True,
+                    raise_if_not_found=False)
+
+                # Add a reference with a tag, to be able to ignore response to this email
+                references = message_dict.get('message_id', '') + ' ' + tools.generate_tracking_message_id('bounce-email')
+                self._routing_create_bounce_email(email_from, body, message, references=references)
+                return True
+
+        return False
+
+    @api.model
+    def _routing_detect_loop_from_headers(self, message):
+        """Detect common headers set by the auto-repliers / mailing lists."""
+
+        # Detect if the email has been sent by an auto-replier or by a mailing list
+        for header in ('X-Autoreply', 'X-Autorespond', 'Feedback-ID', 'List-Id', 'List-Unsubscribe'):
+            if header in message:
+                _logger.info('Email sent by an auto-replier (%s), ignoring it.', header)
+                return True
+
+        if 'Precedence' in message and message['Precedence'].lower() in ('bulk', 'auto_reply', 'list'):
+            _logger.info('Email sent by an auto-replier (Precedence), ignoring it.')
+            return True
+
+        if '-bounce-email@' in message.get('References', ''):
```

Comment:
Can't we use value from msg_dict ? And maybe also use ``in-reply-to`` header, as routing process uses in-reply-to then references ?